### PR TITLE
fix black_temple Teleporter

### PR DESCRIPTION
**Pull request by w5860363**
Imitate the transfer mechanism blizzard in the server。

### CODE REVIEW

File: sql/updates/world/fix black_temple Teleporter.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+UPDATE creature_template SET gossip_menu_id=0,Scriptname = "npc_Spirit_of_Udalo" WHERE entry=23410;
```
Comment (DDuarte) [inline sql/updates/world/fix black_temple Teleporter.sql]: This file does not follow SQL guidelines http://collab.kpsn.org/display/tc/SQL+Guidelines

File: src/server/scripts/Outland/BlackTemple/black_temple.cpp [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
- *
- * This program is distributed in the hope that it will be useful, but WITHOUT
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
- * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
- * more details.
- *
- * You should have received a copy of the GNU General Public License along
- * with this program. If not, see <http://www.gnu.org/licenses/>.
- */
+* Copyright (C) 2008-2015 TrinityCore <http://www.trinitycore.org/>
```
Comment (DDuarte) [inline src/server/scripts/Outland/BlackTemple/black_temple.cpp]: Uneeded space change

File: src/server/scripts/Outland/BlackTemple/black_temple.cpp [hunk 3]
```diff
@@ -29,198 +29,262 @@ Comment:  Spirit of Olum: Player Teleporter to Seer Kanai Teleport after defeati
 
 enum Spells
 {
-    // Spirit of Olum
-    SPELL_TELEPORT                   = 41566,
-    // Wrathbone Flayer
-    SPELL_CLEAVE                     = 15496,
-    SPELL_IGNORED                    = 39544,
-    SPELL_SUMMON_CHANNEL             = 40094
+	// Spirit of Olum
```
Comment (DDuarte) [inline src/server/scripts/Outland/BlackTemple/black_temple.cpp]: Tabs, convert to spaces

File: src/server/scripts/Outland/BlackTemple/black_temple.cpp [hunk 4]
```diff
@@ -29,198 +29,262 @@ Comment:  Spirit of Olum: Player Teleporter to Seer Kanai Teleport after defeati
...
+	EVENT_GET_CHANNELERS = 1,
+	EVENT_SET_CHANNELERS = 2,
+	EVENT_CLEAVE = 3,
+	EVENT_IGNORED = 4,
 };
 
 // ########################################################
 // Spirit of Olum
 // ########################################################
+#define OZ_GOSSIP1          "Take me to the other Deathsworn, Olum."
```
Comment (DDuarte) [inline src/server/scripts/Outland/BlackTemple/black_temple.cpp]: Gossip texts should be in DB

**Comment by DDuarte** [comment]
Re-open the PR once you have fixed the most basic errors.

File: src/server/scripts/Outland/BlackTemple/black_temple.cpp (hunk 4 above)
Comment (w5860363) [inline src/server/scripts/Outland/BlackTemple/black_temple.cpp]: @DDuarte 
blackrock_depths.cpp npc_kharan_mighthammer 
karazhan.cpp npc_barnesAI 

Is to use code Gossip, this is not right?
Please forgive my English

Comment (DDuarte) [inline src/server/scripts/Outland/BlackTemple/black_temple.cpp]: Those scripts are old and have not been updated in a while.
Any new script should follow current standards and at this moment any gossip should be in the database.

Comment (w5860363) [inline src/server/scripts/Outland/BlackTemple/black_temple.cpp]: @DDuarte I understand, I will be in accordance with the new standards and try again the next.

**Pull request closed without merging**